File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        
        max_cnt = 0
        
        for n in s:
            if (n - 1) not in s:
                curr = n
                cnt = 0
                while curr in s:
                    cnt += 1
                    curr +=1
                    max_cnt = max(max_cnt, cnt)
            
        return max_cnt

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.rs
use std::{
    collections::HashSet,
    iter::FromIterator,
    
};

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let mut set : HashSet<i32> = HashSet::from_iter(nums.into_iter());
        
        let mut max_cnt = 0;
        
        while set.len() > 0{
            let n = *set.iter().next().unwrap();
                set.remove(&n);
                
                let mut cnt = 1;
                
                let mut curr = n + 1;
                while set.contains(&curr){
                    curr+=1;
                    cnt +=1;
                    set.remove(&n);
                }
                
                curr = n - 1;
                while set.contains(&curr){
                    curr-=1;
                    cnt+=1;
                    set.remove(&n);
                }
                
                max_cnt = max_cnt.max(cnt);

        }
        
        max_cnt
    }
}